mesonbuild/interpreter/interpreter.py

```diff
@@ -2634,6 +2634,13 @@ def func_configure_file(self, node: mparser.BaseNode, args: T.List[TYPE_var],
 
     def extract_incdirs(self, kwargs, key: str = 'include_directories'):
         prospectives = extract_as_list(kwargs, key)
+        if key == 'include_directories':
+            for i in prospectives:
+                if isinstance(i, str):
+                    FeatureNew.single_use('include_directories kwarg of type string', '0.50.0', self.subproject,
```

Comment:
Why iterating the list twice instead of adding this in the loop below?

Reply:
It felt slightly more convenient that way, because I could break immediately. This should only emit one FeatureNew per handled kwarg, even if that handled kwarg is a list of strings.

I guess it could be a flag variable instead. 
```python
warned = False
for p in prospectives:
    ...
    elif isinstance(p, str):
        result.append(...)
        if not warned and key == 'include_directories':
            FeatureNew.single_use(...)
            warned = True
```

Would you prefer that?

Reply:
Ah, good point about not warning more than one. It's ok like that then.